docs/design/generic.md

```diff
@@ -0,0 +1,94 @@
+# Generic artifact fetching
```

Comment:
**Commit message:**

Please drop the JIRA ID from the commit message, it's an internal non-transparent designation irrelevant to upstream.

Reply:
Dropped